arb/nth_root_pol.c

```diff
@@ -49,11 +49,78 @@ nth_root_pol_prod(acb_t y, acb_srcptr u, const arb_t x, slong d, slong m, slong
 void
 nth_root_pol_turn(acb_t y, acb_srcptr u, const arb_t x, acb_srcptr z, slong d, slong m, slong prec)
 {
-    slong q; /* integer mod 2m */
+  
+    slong q, k, sgn_res;  
+    acb_t s, t;
+    arb_t isgn_s; 
```

Comment:
signs should not be arb type, just int, this will be much simpler
you can use the arb_is_positive &co functions http://arblib.org/arb.html#c.arb_is_positive
in case the sign cannot be determined the function should return an error code (and one should use another nth root)